qubesappmenusext/__init__.py

```diff
@@ -85,6 +85,14 @@ def on_feature_del_appmenus_dispvm(self, vm, event, feature):
         asyncio.ensure_future(self.run_as_user(
             ['qvm-appmenus', '--quiet', '--force', '--update', vm.name]))
 
+    @qubes.ext.handler('domain-feature-pre-set:appmenus-dispvm')
+    def on_feature_pre_set_appmenus_dispvm(self, vm, event, feature,
+            value, oldvalue=None):
+        if value is not None and not vm.template_for_dispvms:
```

Comment:
drop `is not None`, `None` is not the only false value. Especially, features use empty string as canonical false value (but don't hardcode such comparison).